src/targz_sizes.c

```diff
@@ -120,8 +126,24 @@ int main(int argc, char** argv) {
         }
     }
 
+    unsigned int max_filename_blocks;
+    if ( LEVEL_IS_TEST( verbosity ) ) {
+        // when targz_sizes is in *test* mode, set max_filename_blocks
+        // very small (2).  This makes it easier to test the case where
+        // a long filename would overrun the buffer
+        max_filename_blocks=2;
+    } else {
+        // normally we set max_filename_blocks large
+        max_filename_blocks=200;
+    }
+    LOG_INFO(verbosity, "max_filename_blocks=%d", max_filename_blocks );
+
     unsigned char compressed[COMPRESSED_BUFFER];
     unsigned char decompressed[TAR_BLOCK_SIZE];
+    tarblock_t *filename_buff;
+    LOG_INFO(verbosity, "sizeof(tarblock_t)=%d", sizeof(tarblock_t) );
+    filename_buff = (tarblock_t*) calloc(
```

Comment:
OK, my lack of 'C' experience is starting to show through here.  I done my best with this one and it looks good to me.  Addressed in commit: 38ca88b5917f09d66fce33819c7a19d92cd693d2